src/s2wasm.h

```diff
@@ -1347,6 +1348,52 @@ class S2WasmBuilder {
     }
   }
 
+  // Fixes function name hacks caused by LLVM exception & setjmp/longjmp
+  // handling pass for wasm.
+  // This does two things:
+  // 1. Change emscripten_longjmp_jmpbuf to emscripten_longjmp.
+  //    In setjmp/longjmp handling pass in wasm backend, what we want to do is
+  //    to change all function calls to longjmp to calls to emscripten_longjmp.
+  //    Because we replace all calls to longjmp to emscripten_longjmp, the
+  //    signature of that function should be the same as longjmp:
+  //    emscripten_longjmp(jmp_buf, int)
+  //    But after calling a function that might longjmp, while we test whether
+  //    a longjmp occurred, we have to load an int address value and call
+  //    emscripten_longjmp again with that address as the first argument. (Refer
+  //    to lib/Target/WebAssembly/WebAssemblyEmscriptenEHSjLj.cpp in LLVM for
+  //    details.)
+  //    In this case we need the signature of emscripten_longjmp to be (int,
+  //    int). So we need two different kinds of emscripten_longjmp signatures in
+  //    LLVM IR. Both signatures will be lowered to (int, int) eventually, but
+  //    in LLVM IR, types are not lowered yet.
+  //    So we declare two functions in LLVM:
+  //    emscripten_longjmp_jmpbuf(jmp_buf, int)
+  //    emscripten_longjmp(int, int)
+  //    And we change the name of emscripten_longjmp_jmpbuf to
+  //    emscripten_longjmp here.
+  // 2. Converts invoke wrapper names.
+  //    Refer to the comments in fixEmExceptionInvoke below.
+  template<typename ListType>
+  Name fixEmEHSjLjNames(const Name &name, WasmType result,
+                        const ListType &operands) {
+    return fixEmEHSjLjNames(name, getSig(result, operands));
+  }
+
+  Name fixEmEHSjLjNames(const Name &name, const std::string &sig) {
+    if (name == "emscripten_longjmp_jmpbuf")
+      return "emscripten_longjmp";
+    return fixEmExceptionInvoke(name, sig);
+  }
+
+  // This version only converts emscripten_longjmp_jmpbuf and does not deal
+  // with invoke wrappers. This is used when we only have a function name as
+  // relocatable constant.
+  Name fixEmEHSjLjNames(const Name &name) {
```

Comment:
Since this one only handles longjmp, let's give it a different name from the ones that do both longjmp and EH.